src/components/editor/Bindings/OnIncompatibleSchemaChange/Form.tsx

```diff
@@ -0,0 +1,216 @@
+    Typography,
+} from '@mui/material';
+import { useSnackbar } from 'notistack';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useIntl } from 'react-intl';
+import {
+    useFormStateStore_isActive,
+    useFormStateStore_setFormState,
+} from 'stores/FormState/hooks';
+import { snackbarSettings } from 'utils/notification-utils';
+import {
+    useBinding_currentBindingUUID,
+    useBinding_currentCollection,
+} from 'stores/Binding/hooks';
+import { useEditorStore_queryResponse_draftSpecs_schemaProp } from 'components/editor/Store/hooks';
+import { FormStatus } from 'stores/FormState/types';
+import AlertBox from 'components/shared/AlertBox';
+import { autoCompleteDefaultProps } from './shared';
+import useUpdateOnIncompatibleSchemaChange, {
+    BindingMetadata,
+} from './useUpdateOnIncompatibleSchemaChange';
+import useSupportedOptions from './useSupportedOptions';
+import { AutoCompleteOption } from './types';
+import SelectorOption from './SelectorOption';
+
+interface Props {
+    bindingIndex: number;
+}
+
+function OnIncompatibleSchemaChangeForm({ bindingIndex = -1 }: Props) {
+    const intl = useIntl();
+    const { enqueueSnackbar } = useSnackbar();
+
+    const [inputValue, setInputValue] = useState('');
+    const [invalidSetting, setInvalidSetting] = useState(false);
+
+    const currentCollection = useBinding_currentCollection();
+    const currentBindingUUID = useBinding_currentBindingUUID();
+
+    const formActive = useFormStateStore_isActive();
+    const setFormState = useFormStateStore_setFormState();
+
+    const currentSetting = useEditorStore_queryResponse_draftSpecs_schemaProp(
+        bindingIndex,
+        'onIncompatibleSchemaChange'
+    );
+
+    const options = useSupportedOptions();
+    const { updateOnIncompatibleSchemaChange } =
+        useUpdateOnIncompatibleSchemaChange();
+
+    const currentValue = useMemo(() => {
+        if (!currentSetting) {
+            return null;
+        }
+        return options.find((option) => option.val === currentSetting) ?? null;
+    }, [currentSetting, options]);
+
+    const bindingMetadata = useMemo<BindingMetadata[]>(() => {
+        if (bindingIndex > -1 && currentCollection && currentBindingUUID) {
+            return [{ collection: currentCollection, bindingIndex }];
+        }
+
+        return [];
+    }, [bindingIndex, currentBindingUUID, currentCollection]);
+
+    const updateServer = useCallback(
+        async (value?: AutoCompleteOption | null) => {
+            setFormState({ status: FormStatus.UPDATING, error: null });
+
+            updateOnIncompatibleSchemaChange(value?.val, bindingMetadata)
+                .then(() => {
+                    setFormState({ status: FormStatus.UPDATED });
+                })
+                .catch((err) => {
+                    enqueueSnackbar(
+                        intl.formatMessage({
+                            id:
+                                err === 'no binding'
```

Comment:
Highlighting this string in the event you would prefer to store it off into a constant.

Reply:
Naw - leaving as is and later can be shared when we make more of these editors shared.